src/debian.py

```diff
@@ -0,0 +1,68 @@
+import json
+import pathlib
+import subprocess
+from hashlib import sha1
+from os.path import exists
+from subprocess import call
+
+"""Fetch Debian versions with their dates from www.debian.org source repository.
+"""
+
+PRODUCT = "debian"
+REPO_URL = "https://salsa.debian.org/webmaster-team/webwml.git"
+REPO_SHA1 = sha1(REPO_URL.encode()).hexdigest()
+REPO_DIR = pathlib.Path(f"~/.cache/git/debian.md_{REPO_SHA1}").expanduser()
+
+
+# Checkout the Debian website repository.
+def clone_repository():
+    git_opts = f"--git-dir={REPO_DIR}/.git --work-tree={REPO_DIR}"
+    REPO_DIR.mkdir(parents=True, exist_ok=True)
+
+    if not exists(f"{REPO_DIR}/.git"):
+        call(f"git {git_opts} init", shell=True)
+        call(f"git {git_opts} remote add origin {REPO_URL}", shell=True)
+
+    call(f"git {git_opts} config core.sparseCheckout true", shell=True)
+    with open(f"{REPO_DIR}/.git/info/sparse-checkout", "w") as f:
+        f.write("english/News/")
+
+    ret_code = call(f"git {git_opts} pull --depth 1 origin master", shell=True)
+    exit(-ret_code) if ret_code < 0 else None
+
+
+def extract_releases():
+    child = subprocess.Popen(
```

Comment:
Would have preferred if this were in Python instead. But if this feels stable enough, we can merge.

Reply:
I think it is.

I thought about it but reproducing all that commands in Python looked harder ;).